internal/diffgenerator/diffgenerator_test.go

```diff
@@ -0,0 +1,69 @@
+package diffgenerator_test
+
+import (
+	"io"
+	"testing"
+
+	"github.com/cybozu-go/fin/internal/diffgenerator"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestGenerate_Success_EmptyRecords(t *testing.T) {
+	reader, err := diffgenerator.Generate("toSnap", 100*1024*1024)
+	assert.NoError(t, err)
+
+	buf, err := io.ReadAll(reader)
+	assert.NoError(t, err)
+
+	assert.Equal(t, []byte("rbd diff v1\nt\x06\x00\x00\x00toSnaps\x00\x00\x40\x06\x00\x00\x00\x00e"), buf)
+}
+
+func TestGenerate_Success(t *testing.T) {
+	reader, err := diffgenerator.Generate(
+		"toSnap",
+		100*1024*1024,
+		diffgenerator.WithFromSnapName("fromSnap"),
+		diffgenerator.WithHeader("header"),
+		diffgenerator.WithRecords([]diffgenerator.DataRecord{
+			{Type: diffgenerator.RecordTypeUpdated, Offset: 0, Length: 10},
+			{Type: diffgenerator.RecordTypeZero, Offset: 10, Length: 20},
+		}),
+	)
+	assert.NoError(t, err)
+	buf, err := io.ReadAll(reader)
+	assert.NoError(t, err)
+
+	assert.Equal(t, []byte{
+		// HEADER
+		0x68, 0x65, 0x61, 0x64, 0x65, 0x72, // 'h' 'e' 'a' 'd' 'e' 'r'
```

Comment:
Can we rewrite here as follows?

```suggestion
		'h', 'e', 'a', 'd', 'e', 'r',
```